### PR TITLE
Docs: import enterprise documentation

### PR DESCRIPTION
**Pull request by glaubinix**
* Migrate the existing enterprise documentation to the docs repo
* Add a github-action to optimize all png images
* Add a new section about login with email + password

Note the images are located in the folder to make sure we can install the images with the symfony install:assets command.

### CODE REVIEW

File: enterprise/login-email-password-setup.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Login with email and password
```
Comment (naderman) [inline enterprise/login-email-password-setup.md]: Can we rename this to "Authentication with email and password" and the file to "authentication-email-password-setup.md"?

File: enterprise/login-email-password-setup.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+## Private Packagist Enterprise
+
+## Setup
+Navigate to the "Global Configuration" section from the admin page to get to the form below. You can enable login and account creation with password by selecting the option in the form and submitting the form via the "Update configuration" button.
```
Comment (naderman) [inline enterprise/login-email-password-setup.md]: with _a_ password. But better "Enable login and account registration with email addresses and passwords by .."

File: enterprise/login-email-password-setup.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Login with email and password
+## Private Packagist Enterprise
+
+## Setup
```
Comment (naderman) [inline enterprise/login-email-password-setup.md]: Maybe add a line that says, by default authentication with email/password is disabled, and tha we recommend setting up one of the OAuth options only if they are available?

Comment (glaubinix) [reply enterprise/login-email-password-setup.md]: You mean similar to the note on the installation page? `If your source code is hosted on another platform or you do not with to set up an integration then you can also enable [authentication with email and password](./authentication-email-password-setup.md).`

Comment (naderman) [reply enterprise/login-email-password-setup.md]: Except nothing about source code? :S Just what I said above?

File: enterprise/installation.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+## Authentication Setup
+Within Private Packagist Enterprise you now need to set up at least one integration with a third party service for user authentication. You can either use your on-premises version of GitHub, Bitbucket, or GitLab, or their public services on github.com, bitbucket.org or gitlab.com. Follow the instructions to create the respective required identifiers, tokens and secrets.
+
+* [GitHub (Enterprise) Integration Setup](./github-integration-setup.md)
+* [Bitbucket.org Integration Setup](./bitbucket-integration-setup.md)
+* [Bitbucket Server (Stash) Integration Setup](./bitbucket-server-integration-setup.md)
+* [GitLab Integration Setup](./gitlab-integration-setup.md)
+
+If your source code is hosted on another platform or you do not with to set up an integration then you can also enable [authentication with email and password](./authentication-email-password-setup.md).
```
Comment (naderman) [inline enterprise/installation.md]: "wish" not "with".
Can you make this a list item, and make it clearer than you can setup multiple of these but need at least one?

Comment (glaubinix) [reply enterprise/installation.md]: Updated. Right before the list it already says that you need at least one integration.

File: enterprise/login-email-password-setup.md (hunk 3 above)
Comment (glaubinix) [reply enterprise/login-email-password-setup.md]: done

File: enterprise/authentication-email-password-setup.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Authentication with email and password
+## Private Packagist Enterprise
+
+## Setup
+By default authentication with email and password is disabled, and we recommend setting up one of the OAuth options only if they are available.
```
Comment (naderman) [inline enterprise/authentication-email-password-setup.md]: I meant for you to adjust this appropriately, please replace with:

"By default, authentication with email addresses and passwords is disabled. If possible, we recommend setting up authentication with your installation of [GitHub (Enterprise)](./github-integration-setup.md), [Bitbucket.org](./bitbucket-integration-setup.md), [Bitbucket Server (Stash)](./bitbucket-server-integration-setup.md), or [GitLab](./gitlab-integration-setup.md) using OAuth instead."

File: enterprise/installation.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+![HTTP Proxy](/Resources/public/img/docs/enterprise/07-01-http-proxy.png)
+
+In case the HTTP proxy should not be used to access certain domains or IPs then you can configure such
+a NO_PROXY list in the Private Packagist application in the Admin Panel under Global Configuration.
+
+![No HTTP Proxy](/Resources/public/img/docs/enterprise/07-02-http-proxy-no-proxy.png)
+
+## Authentication Setup
+Within Private Packagist Enterprise you now need to set up at least one integration with a third party service for user authentication.
```
Comment (naderman) [inline enterprise/installation.md]: This sentence is no longer true, please replace with something accurate.

File: enterprise/authentication-email-password-setup.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Authentication with email and password
```
Comment (naderman) [inline enterprise/authentication-email-password-setup.md]: sorry this was probably my mistake but I'd use plural passwords and then it should be "Authentication with Email Addresses and Passwords" and same in the filename.

Comment (glaubinix) [reply enterprise/authentication-email-password-setup.md]: updated

File: enterprise/authentication-email-password-setup.md (hunk 5 above)
Comment (glaubinix) [reply enterprise/authentication-email-password-setup.md]: updated

File: enterprise/installation.md (hunk 6 above)
Comment (glaubinix) [reply enterprise/installation.md]: updated

File: enterprise/installation.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+![HTTP Proxy](/Resources/public/img/docs/enterprise/07-01-http-proxy.png)
+
+In case the HTTP proxy should not be used to access certain domains or IPs then you can configure such
+a NO_PROXY list in the Private Packagist application in the Admin Panel under Global Configuration.
+
+![No HTTP Proxy](/Resources/public/img/docs/enterprise/07-02-http-proxy-no-proxy.png)
+
+## Authentication Setup
+Within Private Packagist Enterprise you now need to set up at least one integration with a third party service for user authentication or if your source code is hosted on another platform or you do not wish to set up an integration then you can also enable authentication with email addresses and passwords.
```
Comment (naderman) [inline enterprise/installation.md]: This sentence is too long, the combination of if and multiple ors results in an unclear sentence structure.

Comment (glaubinix) [reply enterprise/installation.md]: Simplified

File: enterprise/installation.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+![HTTP Proxy](/Resources/public/img/docs/enterprise/07-01-http-proxy.png)
+
+In case the HTTP proxy should not be used to access certain domains or IPs then you can configure such
+a NO_PROXY list in the Private Packagist application in the Admin Panel under Global Configuration.
+
+![No HTTP Proxy](/Resources/public/img/docs/enterprise/07-02-http-proxy-no-proxy.png)
+
+## Authentication Setup
+Within Private Packagist Enterprise you now need to set up at least one user authentication method. You have the choice between authentication with email addresses and passwords and several OAuth integrations with a third party service.
```
Comment (naderman) [inline enterprise/installation.md]: integrations with third party services, it's not different integrations for the same service, so the services have to be plural too.

File: enterprise/installation.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+We provide integrations with on-premises version of GitHub, Bitbucket, or GitLab, or their public services on github.com, bitbucket.org or gitlab.com. Follow the instructions to create the respective required identifiers, tokens and secrets.
```
Comment (naderman) [inline enterprise/installation.md]: on-premises version_s_

File: enterprise/authentication-email-password-setup.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# Authentication with Email Addresses and Passwords
```
Comment (naderman) [inline enterprise/authentication-email-password-setup.md]: I did ask you to please also correct the filename again when i asked to update the headline:

enterprise/authentication-email-addresses-passwords-setup.md

Comment (glaubinix) [reply enterprise/authentication-email-password-setup.md]: Missed this. Updated now.

File: enterprise/installation.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+![HTTP Proxy](/Resources/public/img/docs/enterprise/07-01-http-proxy.png)
+
+In case the HTTP proxy should not be used to access certain domains or IPs then you can configure such
+a NO_PROXY list in the Private Packagist application in the Admin Panel under Global Configuration.
+
+![No HTTP Proxy](/Resources/public/img/docs/enterprise/07-02-http-proxy-no-proxy.png)
+
+## Authentication Setup
+Within Private Packagist Enterprise you now need to set up at least one user authentication method. You have the choice between authentication with email addresses and passwords and several OAuth integrations with a third party services.
```
Comment (naderman) [inline enterprise/installation.md]: It now says "with _a_ third party services."

**Pull request merged**